crates/rune-modules/src/http.rs

```diff
@@ -69,26 +69,1046 @@ pub fn module(_stdio: bool) -> Result<Module, ContextError> {
+            /// ```rune,no_run
+            /// use http::StatusCode;
+            ///
+            /// let status_code = StatusCode::GATEWAY_TIMEOUT;
+            /// ```
+        })?;
+
+    module
+        .constant("HTTP_VERSION_NOT_SUPPORTED", 505)
+        .build_associated::<StatusCode>()?
+        .docs(docstring! {
+            /// Status Code: HTTP Version Not Supported
+            ///
+            /// # Examples
+            ///
+            /// ```rune,no_run
+            /// use http::StatusCode;
+            ///
+            /// let status_code = StatusCode::HTTP_VERSION_NOT_SUPPORTED;
+            /// ```
+        })?;
+
+    module
+        .constant("VARIANT_ALSO_NEGOTIATES", 506)
+        .build_associated::<StatusCode>()?
+        .docs(docstring! {
+            /// Status Code: Variant Also Negotiates
+            ///
+            /// # Examples
+            ///
+            /// ```rune,no_run
+            /// use http::StatusCode;
+            ///
+            /// let status_code = StatusCode::VARIANT_ALSO_NEGOTIATES;
+            /// ```
+        })?;
+
+    module
+        .constant("INSUFFICIENT_STORAGE", 507)
+        .build_associated::<StatusCode>()?
+        .docs(docstring! {
+            /// Status Code: Insufficient Storage
+            ///
+            /// # Examples
+            ///
+            /// ```rune,no_run
+            /// use http::StatusCode;
+            ///
+            /// let status_code = StatusCode::INSUFFICIENT_STORAGE;
+            /// ```
+        })?;
+
+    module
+        .constant("LOOP_DETECTED", 508)
+        .build_associated::<StatusCode>()?
+        .docs(docstring! {
+            /// Status Code: Loop Detected
+            ///
+            /// # Examples
+            ///
+            /// ```rune,no_run
+            /// use http::StatusCode;
+            ///
+            /// let status_code = StatusCode::LOOP_DETECTED;
+            /// ```
+        })?;
+
+    module
+        .constant("NOT_EXTENDED", 510)
+        .build_associated::<StatusCode>()?
+        .docs(docstring! {
+            /// Status Code: Not Extended
+            ///
+            /// # Examples
+            ///
+            /// ```rune,no_run
+            /// use http::StatusCode;
+            ///
+            /// let status_code = StatusCode::NOT_EXTENDED;
+            /// ```
+        })?;
+
+    module
+        .constant("NETWORK_AUTHENTICATION_REQUIRED", 511)
+        .build_associated::<StatusCode>()?
+        .docs(docstring! {
+            /// Status Code: Network Authentication Required
+            ///
+            /// # Examples
+            ///
+            /// ```rune,no_run
+            /// use http::StatusCode;
+            ///
+            /// let status_code = StatusCode::NETWORK_AUTHENTICATION_REQUIRED;
+            /// ```
+        })?;
 
-    module.function_meta(Error::string_display)?;
     module.function_meta(StatusCode::string_display)?;
+    module.function_meta(StatusCode::as_integer)?;
```

Comment:
By default, it would be `u16`, but since Rune uses `i64`, I opted for `Integer`.